fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/SearchProviderSpec.groovy

```diff
@@ -526,6 +526,31 @@ abstract class SearchProviderSpec<T extends SearchProvider> extends Specificatio
         resultsPage.numResults == 3
     }
 
+    def "An empty string filter value should return all results"() {
+        setup: "Given a filter that will match every row"
+        /* Expected rows, not necessarily in this order:
+                name: "owl", description: "this is an owl"
+                name: "hawk", description: "this is a raptor"
+                name: "eagle", description: "this is a raptor"
+                name: "kumquat", description: "this is not an animal"
+        */
```

Comment:
Ideally we'd like the code going through the system to preserve or coerce ordering.  Without that we'll be inefficient in terms of creating cache keys.  
I'd suggest a jira ticket to create a Comparator on ApiFilter, probably sorting on DimensionName, DimensionFieldName, OperationName, and then lexically sorted values.  Then the ApiFilters could be stored in a SortedSet, which should make test cases far easier to write.